src/slack/utils.bal

```
@@ -0,0 +1,430 @@
+        return setResError(response);
+    }
+    http:Response resp = <http:Response> response;
+    var jsonResponse = resp.getJsonPayload();
+    if (jsonResponse is error) {
+        return setJsonResError(jsonResponse);
+    }
+    json jsonResp = <json> jsonResponse;
+    var checkOk = check checkOk(jsonResp);       
+}
+
+function getUserIds(http:Client slackClient, string[] users) returns @tainted string|error {
+    string usersList = EMPTY_STRING;
+    var response = slackClient->get(LIST_USERS_PATH);
+    if (response is error) {
+        return setResError(response);
+    }
+    http:Response resp = <http:Response> response;
+    var payload = resp.getJsonPayload();
+    if (payload is error) {
+        return setJsonResError(payload);
+    }
+    map<json> jsonPayload = <map<json>> payload;
+    var checkOk = check checkOk(jsonPayload);
+    var members = check jsonPayload.members;
+    json[] memberList = <json[]> members;
+    foreach var user in users {
+        foreach var member in memberList {
+            if (member.name == user) {
+                usersList = usersList + (member.id).toString() + ",";
+            }
+        }
+    }
+    return usersList;
+}
+
+function getUserId(http:Client slackClient, string user) returns @tainted string|error {
+    string userId = EMPTY_STRING;
+    var response = slackClient->get(LIST_USERS_PATH);
+    if (response is error) {
+        return setResError(response);
+    }
+    http:Response resp = <http:Response> response;
+    var payload = resp.getJsonPayload();
+    if (payload is error) {
+        return setJsonResError(payload);
+    }
+    map<json> jsonPayload = <map<json>> payload;
+    var checkOk = check checkOk(jsonPayload);
+    var members = check jsonPayload.members;
+    json[] memArr = <json[]> members;
+    foreach var member in memArr {
+        if (member.name == user) {
+            userId =  (member.id).toString();
+            break;
+        }
+    }
+    return userId;
```

Isn't it better to return an error (like `UserNotFoundError` or something), if the `userId` is still an `EMPTY_STRING` here?